### PR TITLE
feat: add array_deque for JS and TS

### PR DESCRIPTION
**Pull request by zhuoqinyue**
If this PR is related to coding or code translation, please read the [contribution guideline](https://github.com/krahets/hello-algo/issues/15#issue-1464069591), fill out the checklist, and paste the console outputs to the PR.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

File: codes/javascript/chapter_stack_and_queue/array_deque.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
+/**
+ * File: array_deque.js
+ * Created Time: 2023-02-28
+ * Author: Zhuo Qinyue (1403450829@qq.com)
+ */
+
+
+/* 基于环形数组实现的双向队列 */
+class ArrayDeque {
+    nums;       // 用于存储双向队列元素的数组
```
Comment (krahets) [inline codes/javascript/chapter_stack_and_queue/array_deque.js:10]: Could we set the variable as a private member? Same as others.

File: codes/typescript/chapter_stack_and_queue/array_deque.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
+/**
+ * File: array_deque.ts
+ * Created Time: 2023-02-28
+ * Author: Zhuo Qinyue (1403450829@qq.com)
+ */
+
+
+/* 基于环形数组实现的双向队列 */
+class ArrayDeque {
+    nums: number[];     // 用于存储双向队列元素的数组
```
Comment (krahets) [inline codes/typescript/chapter_stack_and_queue/array_deque.ts:10]: Could we set the variable as a private member? Same as others.

**Comment by krahets** [comment]
Hi @zhuoqinyue ! Could you paste the output of the code, thanks

**Comment by zhuoqinyue** [comment]
OK, I will implement them

**Comment by krahets** [comment]
Hi @zhuoqinyue ! Could you update this PR today? (Waiting to use them in the release, please check the code carefully)

**Comment by zhuoqinyue** [comment]
K sir, I would update it.
I was in trouble last night so couldn't update it in time.
I will mention you when I update it.

**Comment by zhuoqinyue** [comment]
The output of JS:
``` bash
双向队列 deque = [3,2,5]
队首元素 peekFirst = 3
队尾元素 peekLast = 5
元素 4 队尾入队后 deque = [3,2,5,4]
元素 1 队首入队后 deque = [1,3,2,5,4]
队尾出队元素 = 4，队尾出队后 deque = [1,3,2,5]
队首出队元素 = 1，队首出队后 deque = [3,2,5]
双向队列长度 size = 3
双向队列是否为空 = false
```

The output of TS:
``` bash
双向队列 deque = [3,2,5]
队首元素 peekFirst = 3
队尾元素 peekLast = 5
元素 4 队尾入队后 deque = [3,2,5,4]
元素 1 队首入队后 deque = [1,3,2,5,4]
队尾出队元素 = 4，队尾出队后 deque = [1,3,2,5]
队首出队元素 = 1，队首出队后 deque = [3,2,5]
双向队列长度 size = 3
双向队列是否为空 = false

```

**Comment by zhuoqinyue** [comment]
Hey, K sir. I've updated it.

**Review by krahets — APPROVED** [review]
Thanks @zhuoqinyue !

**Pull request merged**